### PR TITLE
Ticket 3787 - Add ioc aldn1000

### PR DESCRIPTION
**Pull request by aaron-long**
Created tests for the WPI Aladdin-1000 Syringe pump

### CODE REVIEW

File: tests/aldn1000.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    @parameterized.expand([('Value 1', 12345), ('Value 2', 1234), ('Value 3', 77424)])
+    @skip_if_recsim("Not supported in RECSIM")
+    def test_GIVEN_new_invalid_high_diameter_WHEN_set_diameter_THEN_diameter_set_limit_returned(self, _, value):
+        invalid_diameter = value
+        expected_diameter = 1000.00
+        self.ca.set_pv_value("DIAMETER:SP", invalid_diameter)
+
+        self.ca.assert_that_pv_is("DIAMETER", expected_diameter, timeout=2)
+
+    @parameterized.expand([('Value 1', -2345), ('Value 2', -1234), ('Value 3', -0676424)])
```
Comment (DominicOram) [inline tests/aldn1000.py]: The superfluous 0 here isn't supported for python 3

File: tests/aldn1000.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+    @parameterized.expand([('Direction 1', 'Withdraw'), ('Direction 2', 'Infuse')])
+    def test_GIVEN_new_direction_WHEN_set_direction_THEN_new_direction_set(self, _, direction):
+        expected_direction = direction
+        self.ca.set_pv_value("DIRECTION:SP", expected_direction)
+
+        self.ca.assert_that_pv_is("DIRECTION", expected_direction)
+
+    @skip_if_recsim("Not supported in RECSIM")
+    def test_GIVEN_existing_direction_WHEN_set_reverse_direction_THEN_direction_reversed(self):
```
Comment (DominicOram) [inline tests/aldn1000.py]: This is only testing one direction (whichever is the default on the emulator) can you make it test both.

File: tests/aldn1000.py (hunk 1 above)
Comment (aaron-long) [reply tests/aldn1000.py]: Removed typo.

File: tests/aldn1000.py (hunk 2 above)
Comment (aaron-long) [reply tests/aldn1000.py]: Created new parameterized test that checks multiple cases for flow direction.

**Pull request merged**